<a href="https://colab.research.google.com/github/DianaBravoPerez/EDP-1/blob/main/interferencia_flecha_cojinete2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##flecha y cojinete

Una flecha se ensambla en un cojinete

Sea $X_1$ el diámetro interior del cojinete y $X_2$ el diámetro de la flecha con

$$X_1 \sim N(1.5,\; 0.0016), \qquad X_2 \sim N(1.48,\; 0.0009).$$


Hay interferencia cuando $X_2 > X_1$

 Se pide la probabilidad de interferencia y el número de réplicas necesarias para estimarla con error menor a $0.01$ y confianza del $95\%$.

## Solución analítica

Sea $D = X_1 - X_2$

 Como $X_1$ y $X_2$ son normales independientes, $D$ es normal con

$$\mu_D = 1.5 - 1.48 = 0.02, \qquad \sigma_D^2 = 0.0016 + 0.0009 = 0.0025.$$

Entonces $P(\text{interferencia}) = P(D < 0) = \phi\!\left( \frac{-0.02}{0.05} \right) = \phi(-0.4).$

In [1]:
import numpy as np
from scipy import stats

mu1, var1 = 1.5, 0.0016
mu2, var2 = 1.48, 0.0009
sigma1 = np.sqrt(var1)
sigma2 = np.sqrt(var2)

mu_D = mu1 - mu2
sigma_D = np.sqrt(var1 + var2)

z = (0 - mu_D) / sigma_D
p_analitica = stats.norm.cdf(z)

print(f"P(interferencia) analítica = {p_analitica:.6f}")

P(interferencia) analítica = 0.344578


## Solución por simulación

Se generan $N$ pares $(x_1, x_2)$ y se cuenta cuántas veces ocurre $x_2 > x_1$

In [2]:
np.random.seed(2025)

N = 100_000
interferencias = 0

for i in range(N):
    x1 = np.random.normal(mu1, sigma1)
    x2 = np.random.normal(mu2, sigma2)
    if x2 > x1:
        interferencias += 1

p_simulada = interferencias / N

print(f"P(interferencia) simulada  = {p_simulada:.6f}")
print(f"P(interferencia) analítica = {p_analitica:.6f}")
print(f"Error absoluto             = {abs(p_simulada - p_analitica):.6f}")

P(interferencia) simulada  = 0.345160
P(interferencia) analítica = 0.344578
Error absoluto             = 0.000582


## Inciso (b): número de réplicas

Se busca $N$ tal que $|\hat{p} - p| < 0.01$ con confianza del $95\%$

Y por el teorema central del límite,

$$N \geq \left( \frac{z_{\alpha/2}}{\varepsilon} \right)^{2} p(1-p),$$

con $z_{0.025} = 1.96$, $\varepsilon = 0.01$ y $p = 0.3446$

In [3]:
epsilon = 0.01
z_alpha = 1.96
p = p_analitica

N_requerido = (z_alpha / epsilon)**2 * p * (1 - p)
N_requerido = int(np.ceil(N_requerido))

print(f"Número de réplicas requerido = {N_requerido}")

Número de réplicas requerido = 8677
